interpreter/text/parser.mly

```diff
@@ -352,10 +336,14 @@ func_type :
       FuncType ($4 c :: ins, out) }
 
 tag_type :
+  | type_use
+    { fun c -> TagType (SynVar ($1 c type_).it, Resumable) }
   | func_type
```

Comment:
Btw, in other places where you can inline a func_type, you can also have both, the type_use _and_ the expanded type. Then desugaring checks that they match up. (Mainly this was introduced so that function parameters can be locally named.)